### PR TITLE
Sir 387 wallet generation updated

### PR DESCRIPTION
**Pull request by OttoVT**

### CODE REVIEW

File: src/VaultAgent.Common/Domain/Wallets/WalletGenerationRequest.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            Id = id;
+            Sequence = sequence;
+            Component = component;
+            VaultId = vaultId;
+            BlockchainId = blockchainId;
+            TenantId = tenantId;
+            CreatedAt = createdAt;
+            UpdatedAt = updatedAt;
+            State = state;
+            Reason = reason;
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent.Common/Domain/Wallets/WalletGenerationRequest.cs:33]: `RejectionReason`

File: src/VaultAgent.Common/Persistence/Wallets/WalletGenerationRequestEntity.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        public string TenantId { get; set; }
+
+        public DateTimeOffset CreatedAt { get; set; }
+
+        public DateTimeOffset UpdatedAt { get; set; }
+
+        public WalletGenerationRequestState State { get; set; }
+
+        public RejectionReason? Reason { get; set; }
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent.Common/Persistence/Wallets/WalletGenerationRequestEntity.cs:27]: RejectionReason

File: src/VaultAgent.Common/Persistence/Wallets/WalletGenerationRequestRepository.cs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                CreatedAt = walletGenerationRequest.CreatedAt,
+                TenantId = walletGenerationRequest.TenantId,
+                UpdatedAt = walletGenerationRequest.UpdatedAt,
+                Sequence = walletGenerationRequest.Sequence,
+                Version = walletGenerationRequest.Version,
+                VaultId = walletGenerationRequest.VaultId,
+                State = walletGenerationRequest.State,
+                RejectionReasonMessage = walletGenerationRequest.RejectionReasonMessage,
+                Component = walletGenerationRequest.Component,
+                BlockchainId = walletGenerationRequest.Component,
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent.Common/Persistence/Wallets/WalletGenerationRequestRepository.cs:132]: `walletGenerationRequest.BlockchainId`

File: src/VaultAgent.Common/Domain/Wallets/WalletGenerationRequest.cs [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            RejectionReasonMessage = rejectionReasonMessage;
+            Version = version;
+        }
+
+        public long Id { get; }
+
+        public long Sequence { get; }
+
+        public string Component { get; }
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent.Common/Domain/Wallets/WalletGenerationRequest.cs:42]: `Context` is missed

File: src/VaultAgent.MessagingContract/Wallets/WalletGenerationRequestUpdated.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace Swisschain.Sirius.VaultAgent.MessagingContract.Wallets
+{
+    public class WalletGenerationRequestUpdated
+    {
+        public long WalletGenerationRequestId { get; set; }
+
+        public long Sequence { get; set; }
+
+        public string Component { get; set; }
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent.MessagingContract/Wallets/WalletGenerationRequestUpdated.cs:11]: `Context` missed

File: src/VaultAgent.MessagingContract/Wallets/WalletGenerationRequestUpdated.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+        public string TenantId { get; set; }
+
+        public DateTime CreatedAt { get; set; }
+
+        public DateTime UpdatedAt { get; set; }
+
+        public WalletGenerationRequestState State { get; set; }
+
+        public RejectionReason? Reason { get; set; }
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent.MessagingContract/Wallets/WalletGenerationRequestUpdated.cs:25]: `RejectionReason`

File: src/VaultAgent/GrpcServices/VaultsService.cs [hunk 7]
```diff
@@ -45,7 +45,7 @@ request.Type switch
                         _ => throw new ArgumentOutOfRangeException(nameof(request.Type), request.Type, null)
                     });
 
-                await _vaultsRepository.AddOrIgnoreAsync(vault);
+                await _vaultsRepository.AddAsync(vault);
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent/GrpcServices/VaultsService.cs:48]: Why this was changed?

File: src/VaultAgent/GrpcServices/WalletsService.cs [hunk 8]
```diff
@@ -15,66 +19,137 @@ public class WalletsService : Wallets.WalletsBase
...
+
+                var walletGenerationRequest = 
+                    WalletGenerationRequest.Create(
+                        outbox.AggregateId,
+                        request.Component,
+                        request.VaultId,
+                        request.BlockchainId,
+                        request.TenantId);
+
+                await _walletGenerationRequestRepository.AddAsync(walletGenerationRequest);
```
Comment (KonstantinRyazantsev) [inline src/VaultAgent/GrpcServices/WalletsService.cs:92]: Why not `AddOrIgnore`?

Comment (OttoVT) [reply src/VaultAgent/GrpcServices/WalletsService.cs:92]: this is a place that should be executed once in order to keep the system in a consistent state.

File: src/VaultAgent/GrpcServices/VaultsService.cs (hunk 7 above)
Comment (OttoVT) [reply src/VaultAgent/GrpcServices/VaultsService.cs:48]: AddOrIgnoreAsync is a wrong method for adding in service.

**Pull request merged**